http.go

```diff
@@ -407,17 +406,15 @@ func (h *MockHandler) ServerHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if h.StatusCode > 0 {
+		w.WriteHeader(h.StatusCode)
+	}
+
 	if h.Body != nil {
 		if _, err := w.Write(h.Body); err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
 			h.writeError = err
-			return
 		}
 	}
-
-	if h.StatusCode > 0 {
-		w.WriteHeader(h.StatusCode)
-	}
 }
 
 // WriteError returns the most recent error from writing to the
```

Comment:
Small nit: thinking about it, `WriteError` sounds like you're writing an error and not getting the error that happening during writing, can this be phrased as something else like `GetWriteError`?